feat(progress): Add interrupt option

jag3dagster opened the pull request:
Add the ability to specify if an action can be interrupted. Options that I've thought of off the bat are ped going ragdoll or ped being handcuffed.

thelindat commented on resource/interface/client/progress.lua:
```diff
@@ -40,32 +40,48 @@ local function startProgress(data)
-				DisableControlAction(0, 25, true)
-				DisablePlayerFiring(cache.playerId, true)
+			if data.interrupt then
+				if data.interrupt.ragdoll then
+					if IsPedRagdoll(cache.ped) then
+						progress = false
+					end
+				end
+
+				if data.interrupt.cuff then
+					if IsPedCuffed(cache.ped) then
+						progress = false
+					end
+				end
```
```lua
if data.interrupt.ragdoll and IsPedRagdoll(cache.ped) then
    progress = false
end

if data.interrupt.cuff and IsPedCuffed(cache.ped) then
    progress = false
end
```

As above would be better. These would likely also be better off as defaults (so you have to explicitly set them to false to disable), and should get checked before the progress bar begins.

jag3dagster replied:
Updated based on feedback

jag3dagster commented:
Ope, realized I overlooked something... You can hold off on the review for a sec

jag3dagster commented:
Alright, fixed what I overlooked and added falling to potential interrupts to ignore.

Interesting thing I realized during testing of this though was that ox_inventory useItem is already checking for dead / ragdoll / falling before attempting to use an item. Would y'all want that updated to be a check for ignoreInterrupt in client table defaulting to current behavior if no ignore is found?
